Project/core/src/main/java/Controller/CheckAction.java

```diff
@@ -170,6 +183,22 @@ private List<int[]> mindSlip(Recruiter player, Board board, int type) {
 
     }
 
+    /**
+     * Gets valid moves for a player, works for both recruiter and rougeagents
+     * 
+     * @param player player that is being considered
+     * @param board  Board that the player inhabits
+     * @return Returns a mask that contains booleans saying where a player
+     *         can move
+     */
+    public boolean[][] getValidMoves(Player player, Board board) {
```

Comment:
Same here, both movement branch and the movement issue will conflict. Can be solved later though